packages/babel-generator/src/node/index.ts

```diff
@@ -80,43 +79,40 @@ function isOrHasCallExpression(node: t.Node): boolean {
 export function needsWhitespace(
   node: t.Node,
   parent: t.Node,
-  type: "before" | "after",
+  type: number,
```

Comment:
Ok, I see that you cannot export `WhitespaceFlag` from the other file because it gets deopted. I opened https://github.com/babel/babel/pull/14723 to fix it, so you can use `export type { WhitespaceFlag }` (we can already do it in this PR, and update our `@babel/` dependencies as soon as #14723 gets released).

Reply:
```ts
export function needsWhitespaceBefore(node: t.Node, parent: t.Node) {
  return needsWhitespace(node, parent, 1);
}

export function needsWhitespaceAfter(node: t.Node, parent: t.Node) {
  return needsWhitespace(node, parent, 2);
}
```

But we can still only use 1 and 2 here, because type imports cannot be used as values.

Reply:
Yes, but at least the type annotation will help.

Reply:
Unfortunately it still doesn't seem to be possible to optimize.😰

Reply:
Ah, it was because I didn't update the dependencies. Should I change @babel/preset-typescript to workspace:^?

Reply:
No, because we can't built Babel with Babel itself unless Babel is already built :stuck_out_tongue: 

I'll release a patch before merging this, then we can merge this PR and (in a separate PR) update our `@babe/*` dependencies.